File: core/src/projection/core/Asset.cpp
#include "projection/core/Asset.h"

namespace projection::core {

Asset::Asset(AssetId id, std::string name, AssetType type, std::string path, std::vector<AssetVariant> variants)
    : id_(std::move(id)),
      name_(std::move(name)),
      type_(type),
      path_(std::move(path)),
      variants_(std::move(variants)) {}

}  // namespace projection::core

File: clients/composer/src/types/asset.ts
export type AssetType = "VideoFile" | "ImageFile";

export type AssetVariant = {
  path: string;
  note: string;
};

export type Asset = {
  id: string;
  name: string;
  type: AssetType;
  path: string;
  variants: AssetVariant[];
};

